src/app/components/SearchBar/SearchBar.tsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import styled from 'styled-components';
+import SearchIcon from '../Icons/SearchIcon';
+
+type SearchBarProps = {
+  searchQuery: string;
+  setSearchQuery: (searchQuery: string) => void;
+  onSubmit: () => void;
+  placeholderText: string;
+  screenreaderLabelText: string;
+};
+
+export default function SearchBar({
+  searchQuery,
+  setSearchQuery,
+  onSubmit,
+  placeholderText,
+  screenreaderLabelText,
+}: SearchBarProps): JSX.Element {
+  return (
+    <FormContainer
+      onSubmit={(event) => {
+        event.preventDefault();
+        onSubmit();
+        console.log(event);
+      }}
+    >
+      <StyledButton type="submit">
+        <SearchIcon height="40" width="40"></SearchIcon>
+      </StyledButton>
+      <ScreenReaderOnlyLabel htmlFor="styledInput">
+        {screenreaderLabelText}
+      </ScreenReaderOnlyLabel>
+      <StyledInput
+        id="styledInput"
+        type="search"
+        placeholder={placeholderText}
+        value={searchQuery}
+        onChange={(event) => setSearchQuery(event.target.value)}
+      />
```

Comment:
To make sure that `id` are unique, you could cascade label and input.

```suggestion
     <label>
        <ScreenReaderOnly>
          {screenreaderLabelText}
        </ScreenReaderOnly>
        <StyledInput
          id="styledInput"
          type="search"
          placeholder={placeholderText}
          value={searchQuery}
          onChange={(event) => setSearchQuery(event.target.value)}
        />
      </label>
```